src/common/BaseLightAccount.sol

```diff
@@ -123,10 +157,12 @@ abstract contract BaseLightAccount is BaseAccount, TokenCallbackHandler, UUPSUpg
     }
 
     function _call(address target, uint256 value, bytes memory data) internal {
-        (bool success, bytes memory result) = target.call{value: value}(data);
-        if (!success) {
-            assembly ("memory-safe") {
-                revert(add(result, 32), mload(result))
+        assembly ("memory-safe") {
+            let succ := call(gas(), target, value, add(data, 32), mload(data), 0, 0)
+            if iszero(succ) {
+                // We can overwrite memory since we're going to revert out of this call frame anyway
+                returndatacopy(0, 0, returndatasize())
+                revert(0, returndatasize())
```

Comment:
Technically, this can lead to undefined behavior: [solidity docs](https://docs.soliditylang.org/en/v0.8.24/assembly.html#memory-safety:~:text=Since%20this%20is%20mainly%20about%20the%20optimizer%2C%20these%20restrictions%20still%20need%20to%20be%20followed%2C%20even%20if%20the%20assembly%20block%20reverts%20or%20terminates.).

I know we could check the compiler output to see if anything weird is happening, but for ease of maintainability, I think we should be storing this starting at the free mem pointer (`mload(0x40)`) rather than 0. We should be ok not updating the free mem ptr though, since it will revert afterwards and we're not doing memory operations in between.

Reply:
Makes sense-- good catch!

I think we should keep all blocks memory safe regardless, I don't think I'll be able to beat the optimizer in the long run-- and I want us to avoid headaches, I'll go ahead and use the free memory pointer for the sake of maintainability.

Reply:
Is the concern the possibility that the stack parameters needed to get to the `revert` opcode may be loaded in memory prior to `returndatacopy`? That's an interesting problem, and definitely possible (likely?) in the future.

Reply:
To be honest, I've yet to see an actual case where this causes an issue, but yes hypothetically things could be rearranged by the optimizer such that an access of a stack variable lifted to memory results in undefined behavior after a not-memory-safe operation.